healthcheck/healthcheck_test.go

```diff
@@ -524,3 +524,39 @@ func TestCheckTimeout(t *testing.T) {
 		t.Errorf("Expected state change notification not received")
 	}
 }
+
+func TestCheckDryrun(t *testing.T) {
+	notify := make(chan *Notification, 10)
+	hc := NewCheck(notify)
+	hc.Blocking(true)
+	hc.Dryrun(true)
+	go hc.Run(nil)
+
+	config := NewConfig(1, &fakeChecker{succeed: false})
+	config.Interval = 250 * time.Millisecond
+	hc.Update(config)
+
+	time.Sleep(800 * time.Millisecond)
```

Comment:
Thanks for having better test coverage. I think you are testing too much here. You can simply wait on the health results (<-notify) and make sure it's healthy. The rest is not changed and covered by DryRun

Reply:
You're right. Just sent another update. Pls take a look.